torch/testing/_internal/schema_check_mode.py

```diff
@@ -37,9 +44,17 @@ def has_mutated(before, after):
         for argument in func._schema.arguments:
             name = argument.name if argument.name != "self" else "input"
             if arguments.get(name) is not None:
-                before = tree_flatten(arguments.get(name))[0]
-                after = tree_flatten(cloned_arguments.get(name))[0]
+                before = tree_flatten(cloned_arguments.get(name))[0]
+                after = tree_flatten(arguments.get(name))[0]
                 if (any([has_mutated(i, j) for i, j in zip(before, after)]) and not argument.is_mutable):
                     raise RuntimeError(f"Argument {name} is not defined as mutable but was mutated")
+                for i in before:
```

Comment:
very minor nit: i feel like `i` is usually the name of an index, maybe rename it `b`? totally not important though, you can ignore if you don't want to change it.